2-Operating-Systems/1-arch.md

```diff
@@ -0,0 +1,459 @@
+     nitrogen         `# wallpaper manager` \
+     slock            `# simple lock screen` \
+     xorg             `# graphical user interface infrastructure` \
+     xorg-xinit       `# allow starting X from command line` \
+     xf86-video-intel `# graphics driver for Intel chipsets` \
+     ccid             `# CCID driver for some smartcards` \
+     opensc           `# OpenSC driver for some smartcards` \
+     openssh          `# Secure shell into other computers` \
+     haveged          `# More secure random numbers` \
+     pulseaudio       `# Audio support` \
+     pulseaudio-alsa  `# ALSA audio driver support` \
+     pulsemixer       `# Console audio manager`
+    ```
+    
+    Note: Non-Intel graphics users, see: [Arch Driver Installation][2]
+
+13. Fstab: Generate 'File System TABle' of contents
+    
+    ```
+    genfstab -U /mnt >> /mnt/etc/fstab
+    ```
+
+14. Chroot: login to new arch installation by changing root
+    
+    ```
+    arch-chroot /mnt
+    ```
+
+15. Initramfs: install system initialization bundle
+    
+    ```
+    # Enable encryption support in initramfs
+    sed -i 's/ filesystems/ encrypt filesystems/g' /etc/mkinitcpio.conf
+    
+    # Generate new initramfs
+    mkinitcpio -p linux
+    ```
+
+16. Configure and install bootloader
+    
+    ```
+    bootctl --path=/boot install
+    echo "default arch" > /boot/loader/loader.conf
+    echo "title Arch Linux" >> /boot/loader/entries/arch.conf
+    echo "linux /vmlinuz-linux" >> /boot/loader/entries/arch.conf
+    echo "initrd /initramfs-linux.img" >> /boot/loader/entries/arch.conf
+    echo "options cryptdevice=UUID=$(blkid -o value /dev/mmcblk0p2 \
+    | head -n1):cryptroot root=/dev/mapper/cryptroot rw" >> \
+    /boot/loader/entries/arch.conf
+    ```
+
+17. Configure system time keeping
+    
+    ```
+    timedatectl set-ntp true
+    timedatectl set-timezone Region/City
+    hwclock --systohc
+    ```
+
+18. Localization: setting your preferred language
+    
+    ```
+    # Uncomment preferred language in /etc/locale.gen
+    sed -i 's/^#en_US/en_US/g' /etc/locale.gen
+    
+    # Create /etc/locale.conf and set default language
+    echo "LANG=en_US.UTF-8" > /etc/locale.conf
+    
+    # Non-US keyboard users can set their default layout
+    echo "KEYMAP=de-latin1" > /etc/vconsole.conf
+    
+    # Re-generate locales
+    locale-gen
+    ```
+
+19. If needed: Set rotation
+    
+    ```
+    sed -i '$s/$/ fbcon=rotate:1/' /boot/loader/entries/arch.conf
+    ```
+
+20. If needed: Set font
+    
+    ```
+    echo "FONT=latarcyrheb-sun32" >> /etc/vconsole.conf
+    ```
+
+21. Create a user with super user rights
+    
+    ```
+    echo "%wheel ALL=(ALL) ALL" >> /etc/sudoers
+    useradd -m -G wheel -s /bin/bash janedoe
+    passwd janedoe
+    ```
+
+22. Name your computer!
+    
+    ```
+    # Create the hostname file
+    hostnamectl set-hostname computername
+    
+    # Specify new hostname for the network
```

Comment:
FWIW I don't bother with this.

Reply:
Why not just `echo 'HOSTNAME' > /etc/hostname`?